tests/console/sssd_389ds_functional.pm

```diff
@@ -0,0 +1,114 @@
+        assert_script_run("curl " . data_url("sssd/398-ds/sssd.conf") . " -o /etc/sssd/sssd.conf");
+        assert_script_run("curl " . data_url("sssd/398-ds/nsswitch.conf") . " -o /etc/nsswitch.conf");
+        assert_script_run("curl " . data_url("sssd/398-ds/config") . " --create-dirs -o ~/.ssh/config");
+        systemctl("disable --now nscd.service");
+        systemctl("enable --now sssd.service");
+        set_var("SSSD_389DS_FUNCTIONAL_SETUP", 1);
+    }    # setup test
+
+
+    #execute test cases
+    #get remote user indentity
+    validate_script_output("id alice", sub { m/uid=9998\(alice\)/ });
+    #remote user authentification test
+    assert_script_run("pam-config -a --sss --mkhomedir");
+    validate_script_output('sshpass -p open5use ssh mary@localhost whoami', sub { m/mary/ });
+    #Change password of remote user
+    assert_script_run('sshpass -p open5use ssh alice@localhost \'echo -e "open5use\nn0vell88\nn0vell88" | passwd\'');
+    validate_script_output('sshpass -p n0vell88 ssh alice@localhost echo "login as new password!"', sub { m/new password/ });
+    validate_script_output('ldapwhoami -x -H ldap://ldapserver -D uid=alice,ou=users,dc=sssdtest,dc=com -w n0vell88', sub { m/alice/ }); #verify password changed in remote 389-ds.
+                                                                                                                                         #Sudo run a command as another user
+    assert_script_run("sed -i '/Defaults targetpw/s/^/#/' /etc/sudoers");
+    validate_script_output('sshpass -p open5use ssh mary@localhost "echo open5use|sudo -S -l"', sub { m#/usr/bin/cat# });
+    assert_script_run(qq(su -c 'echo "file read only by owner alice" > hello && chmod 600 hello' -l alice));
+    validate_script_output('sshpass -p open5use ssh mary@localhost "echo open5use|sudo -S -u alice /usr/bin/cat /home/alice/hello"', sub { m/file read only by owner alice/ });
+    #Change back password of remote user
+    assert_script_run('sshpass -p n0vell88 ssh alice@localhost \'echo -e "n0vell88\nopen5use\nopen5use" | passwd\'');
+    validate_script_output('sshpass -p open5use ssh alice@localhost echo "Password changed back!"', sub { m/Password changed back/ });
+
+    #offline identity lookup and authentification
+    assert_script_run("$docker stop ds389_container") if ($docker eq "docker");
+    #offline cached remote user indentity lookup
+    validate_script_output("id alice", sub { m/uid=9998\(alice\)/ });
+    #offline remote user authentification test
+    validate_script_output('sshpass -p open5use ssh mary@localhost whoami', sub { m/mary/ });
+    #offline sudo run a command as another user
+    validate_script_output('sshpass -p open5use ssh mary@localhost "echo open5use|sudo -S -u alice /usr/bin/cat /home/alice/hello"', sub { m/file read only by owner alice/ });
+}
+#rollback to clean up test enviroment
```

Comment:
Roollback or not, if not then remove the commented lines

Reply:
No roollback, Just removed them.